tests/integration/pilot/sidecar_api_test.go

```diff
@@ -72,7 +70,7 @@ func TestSidecarListeners(t *testing.T) {
 	}
 
 	// Test the empty case where no config is loaded
-	err = pilotInst.WatchDiscovery(time.Second*30,
+	err = pilotInst.WatchDiscovery(time.Second * 10,
```

Comment:
Note that 10 seconds may still not be enough in CI.
I've added partial time update support, like here:
https://github.com/istio/istio/blob/f5e4f745a136b78400dd41c4dd6285ccdecfffa4/pkg/test/framework/components/istio/config.go#L158

I think it makes sense to rationalize and have a single model for such timeouts over time.

Reply:
Lets see what happens in CI. In my local test this passed in 2s with the signal change whereas previously it was close to 10s. Will look at the CIMode thing.